test/pkg/scm/scm.go

```diff
@@ -48,7 +97,12 @@ func gitPushPullRetry(t *testing.T, opts *Opts, path string) {
 			opts.Log.Infof("Pushed files to repo %s branch %s", opts.WebURL, opts.TargetRefName)
 			// trying to avoid the multiple events at the time of creation we have a sync
 			time.Sleep(5 * time.Second)
-			return
+
+			// get sha
+			sha, err := git.RunGit(path, "rev-parse", "HEAD")
```

Comment:
Moving the `rev-parse` to here as this function may change the sha if it performs a rebase